packages/@react-aria/test-utils/src/select.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, waitFor, within} from '@testing-library/react';
+import {BaseTesterOpts, UserOpts} from './user';
+
+export interface SelectOptions extends UserOpts, BaseTesterOpts {
+  // TODO: I think the type grabbed from the testing library dist for UserEvent is breaking the build, will need to figure out a better place to grab from
+  user: any
+}
+export class SelectTester {
+  private user;
+  private _interactionType: UserOpts['interactionType'];
+  private _trigger: HTMLElement | undefined;
+
+  constructor(opts: SelectOptions) {
+    let {root, user, interactionType} = opts;
+    this.user = user;
+    this._interactionType = interactionType || 'mouse';
+    // Handle case where the wrapper element is provided rather than the Select's button (aka RAC)
+    let triggerButton = within(root).queryByRole('button');
+    if (triggerButton == null) {
+      triggerButton = root;
+    }
+    this._trigger = triggerButton;
+  }
+
+  setInteractionType = (type: UserOpts['interactionType']) => {
+    this._interactionType = type;
+  };
+
+  open = async (opts: {interactionType?: UserOpts['interactionType']} = {}) => {
+    let {
+      interactionType = this._interactionType
+    } = opts;
+    let trigger = this.trigger;
+    let isDisabled = trigger.hasAttribute('disabled');
+
+    if (interactionType === 'mouse') {
+      await this.user.click(this._trigger);
+    } else if (interactionType === 'keyboard') {
+      act(() => trigger.focus());
+      await this.user.keyboard('[Enter]');
```

Comment:
Good point, I think there is value for adding options for specific interaction types. @snowystinger actually did so for menu here: https://github.com/adobe/react-spectrum/pull/7011/files#diff-57b168b95fed0b30ba5fd5fd9c08ddbea2fa2f475ba1041388225097c3d5624e, we can definitely iterate on this. I was initially targeting app scale tests and users who wanted just common interactions rather than component library type tests but definitely can do both

Reply:
we can add to these apis down the line as tests need them I think, it's really hard to predict ahead of time, there's a lot of interactions to remember for each component